### ISSUE TITLE
Problems with using uiGrid (or anything that is using "Scope" in a string in the controller)

### ISSUE THREAD

**Issue by magbeat** [Urigo/angular-meteor#943]
We are getting the following error when using the 'Scope' string anywhere in our controller: 

```
debug.js:41 Exception in queued task: TypeError: this.scope.$digest is not a function
    at ReactiveContext._propertyChanged (http://localhost:3000/packages/angular-meteor-data.js?824d3de5440cdbff4fbb313ba44222b234957eca:2390:22)
    at Object.cursor.observe.addedAt (http://localhost:3000/packages/angular-meteor-data.js?824d3de5440cdbff4fbb313ba44222b234957eca:2263:21)
    at observeChangesCallbacks.addedBefore (http://localhost:3000/packages/minimongo.js?cdf1a26cf7719fa9471a8017c3defd5aea812727:3623:28)
    at Object.self.applyChange.addedBefore (http://localhost:3000/packages/minimongo.js?cdf1a26cf7719fa9471a8017c3defd5aea812727:3560:56)
    at http://localhost:3000/packages/minimongo.js?cdf1a26cf7719fa9471a8017c3defd5aea812727:407:13
    at _.extend.runTask (http://localhost:3000/packages/meteor.js?9730f4ff059088b3f7f14c0672d155218a1802d4:685:11)
    at _.extend.flush (http://localhost:3000/packages/meteor.js?9730f4ff059088b3f7f14c0672d155218a1802d4:713:10)
    at _.extend.drain (http://localhost:3000/packages/meteor.js?9730f4ff059088b3f7f14c0672d155218a1802d4:721:12)
    at LocalCollection.insert (http://localhost:3000/packages/minimongo.js?cdf1a26cf7719fa9471a8017c3defd5aea812727:615:22)
    at Object.self._connection.registerStore.update (http://localhost:3000/packages/mongo.js?69942a86515ec397dfd8cbb0a151a0eefdd9560d:250:28)
```

We are using the controllerAs syntax and therefore:

```
$reactive(vm).attach($scope);
```

The constructor of ReactiveContext in the built angular-meteor-data.js file consists of the following line:

```
if ((this.context.constructor || angular.noop).toString().indexOf('Scope') > -1) {
        this.scope = this.context;  
}
```

On the other hand we are using uiGrid and the ng-click attribute in the controller:
```
... ng-click="grid.appScope.vm.navigateToProperty(propertyId)" ...
```

The condition evaluates to true and sets ```this.scope``` to ```this.context``` which should not happen.

Later on when we call ```attach```
```
if (!this.scope && (scope.constructor || angular.noop).toString().indexOf('Scope') > -1) {
  this.scope = scope;
}
```
which evaluates to false because of ```!this.scope```and therefore ```this.scope``` is not set correctly.

**Comment by canac** [comment]
I am running into this exception as well. I don't have a firm grasp on what the code is trying to do, but if the goal of that test is to check whether `this` is an AngularJS scope, maybe instead of testing `(this.context.constructor || angular.noop).toString().indexOf('Scope') > -1` you could test `(this.context.constructor || angular.noop) === $rootScope.constructor`.

**Comment by canac** [comment]
Looking through the code, it appears that this may have already been fixed in #923 (which did not make it into the 1.3 release). So hopefully this particular exception will disappear in the 1.3.1 release.

**Comment by magbeat** [comment]
@canac that would explain why I could not find the code in question when checked out from Github. Thanks for the info!

**Comment by dotansimha** [comment]
@magbeat 
Sorry for the inconvenience, AS @canac  said, it already been fixes, and will be released soon in 1.3.1

**Comment by magbeat** [comment]
@dotansimha thanks for the info, the fix and your work on the project. I'll wait for the fix in 1.3.1 then.

**Comment by Urigo** [comment]
Fixed in the new [1.3.1 release](https://github.com/Urigo/angular-meteor/releases)
